pkg/storaged/content-views.jsx

```diff
@@ -960,12 +1121,33 @@
                            }
                        }),
         ],
+        update: (dlg, vals, trigger) => {
+            if (vals.purpose == 'block' && (trigger == "layout" || trigger == "pvs" || trigger == "purpose")) {
+                const max = max_size(vals);
+                const old_max = dlg.get_options("size").max;
+                if (vals.size > max || vals.size == old_max)
+                    dlg.set_values({ size: max });
+                dlg.set_options("size", { max });
+                dlg.set_options("layout", { explanation: layout_descriptions[vals.layout] });
+            } else if (trigger == "purpose") {
+                dlg.set_options("size", { max: vgroup.FreeSize });
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17226-20230831-122253-decca6bc-fedora-39-storage//Coverage/lcov/github-pr.diff.gcov.html)